datatable/java/datatable/src/main/java/io/cucumber/datatable/DataTableTypeRegistry.java

```diff
@@ -26,69 +26,79 @@
-        defineDataTableType(new DataTableType(optionalBigInteger, transformAsOptional(bigIntegerTableCellTransformer)));
+        {
+            TableCellTransformer<Object> objectTableCellTransformer = applyIfPresent(s -> s);
+            defineDataTableType(new DataTableType(Object.class, objectTableCellTransformer, true));
+            defineDataTableType(new DataTableType(Optional.class, optionalIfAbsent(objectTableCellTransformer), true));
+        }
+        {
+            TableCellTransformer<Object> stringTableCellTransformer = applyIfPresent(s -> s);
+            defineDataTableType(new DataTableType(String.class, stringTableCellTransformer, true));
+            Type optionalString = new TypeReference<Optional<String>>() {}.getType();
+            defineDataTableType(new DataTableType(optionalString, optionalIfAbsent(stringTableCellTransformer), true));
+        }
+        {
+            TableCellTransformer<BigInteger> bigIntegerTableCellTransformer = applyIfPresent(BigInteger::new);
+            defineDataTableType(new DataTableType(BigInteger.class, bigIntegerTableCellTransformer));
+            Type optionalBigInteger = new TypeReference<Optional<BigInteger>>() {}.getType();
+            defineDataTableType(new DataTableType(optionalBigInteger, optionalIfAbsent(bigIntegerTableCellTransformer)));
+        }
+        {
+            TableCellTransformer<BigDecimal> bigDecimalTableCellTransformer = applyIfPresent(numberParser::parseBigDecimal);
+            defineDataTableType(new DataTableType(BigDecimal.class, bigDecimalTableCellTransformer));
+            Type optionalBigDecimal = new TypeReference<Optional<BigDecimal>>() {}.getType();
+            defineDataTableType(new DataTableType(optionalBigDecimal, optionalIfAbsent(bigDecimalTableCellTransformer)));
+        }
+        {
+            TableCellTransformer<Byte> byteTableCellTransformer = applyIfPresent(Byte::decode);
+            defineDataTableType(new DataTableType(Byte.class, byteTableCellTransformer));
+            defineDataTableType(new DataTableType(byte.class, byteTableCellTransformer));
+            Type optionalByte = new TypeReference<Optional<Byte>>() {}.getType();
+            defineDataTableType(new DataTableType(optionalByte, optionalIfAbsent(byteTableCellTransformer)));
+        }
+        {
+            TableCellTransformer<Short> shortTableCellTransformer = applyIfPresent(Short::decode);
+            defineDataTableType(new DataTableType(Short.class, shortTableCellTransformer));
+            defineDataTableType(new DataTableType(short.class, shortTableCellTransformer));
+            Type optionalShort = new TypeReference<Optional<Short>>() {}.getType();
+            defineDataTableType(new DataTableType(optionalShort, optionalIfAbsent(shortTableCellTransformer)));
+        }
+        {
+            TableCellTransformer<Integer> integerTableCellTransformer = applyIfPresent(Integer::decode);
```

Comment:
I've moved the optional transformers next to their non-optional counter parts. This allowed me to see that the transformer for integer was missing.